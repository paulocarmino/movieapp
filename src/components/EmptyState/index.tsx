import Link from "next/link"
import Badge from "../Badge"

const suggestions = [
  { name: 'Minions', href: '/?search=Minions' },
  { name: 'Captain Marvel', href: '/?search=Captain%20Marvel' },
  { name: 'Rambo', href: '/?search=Rambo' },
  { name: 'Mission Impossible', href: '/?search=Mission%20Impossible' },
  { name: 'Spiderman', href: '/?search=Spiderman' }
]

const EmptyState = () => {
  return (
    <div className="flex flex-col items-center my-10">
      <svg xmlns="http://www.w3.org/2000/svg" className="mx-auto w-24 h-24 text-gray-100" fill="none" viewBox="0 0 24 24" stroke="currentColor" strokeWidth="1">
        <path strokeLinecap="round" strokeLinejoin="round" d="M10 21h7a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v11m0 5l4.879-4.879m0 0a3 3 0 104.243-4.242 3 3 0 00-4.243 4.242z" />
      </svg>

      <h3 className="mt-2 text-base font-bold text-gray-100">Do a search!</h3>
      <p className="mt-1 text-sm text-gray-400">Search for a movie above, or try any of the suggestions below:</p>

      <div className="mt-6 w-full text-center lg:max-w-[500px]">
        {suggestions.map((suggestion, index) => (
          <Badge key={index}>
            <Link href={suggestion.href}>
              {suggestion.name}
            </Link>
          </Badge>
        ))}
      </div>
    </div>
  )
}
export default EmptyState
