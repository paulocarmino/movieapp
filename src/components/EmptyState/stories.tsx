import { Story, Meta } from '@storybook/react'
import EmptyState from '.'

export default {
  title: 'EmptyState',
  component: EmptyState,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  }
} as Meta

export const Default: Story = () => <EmptyState />
