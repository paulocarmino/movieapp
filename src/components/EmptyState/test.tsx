import { render, screen } from '@testing-library/react'

import EmptyState from '.'

describe('<EmptyState />', () => {
  it('should render the heading', () => {
    const { container } = render(<EmptyState />)

    expect(screen.getByRole('heading', { name: /Do a search!/i })).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
