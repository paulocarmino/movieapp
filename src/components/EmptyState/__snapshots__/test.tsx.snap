// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> should render the heading 1`] = `
<div
  class="flex flex-col items-center my-10"
>
  <svg
    class="mx-auto w-24 h-24 text-gray-100"
    fill="none"
    stroke="currentColor"
    stroke-width="1"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 21h7a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v11m0 5l4.879-4.879m0 0a3 3 0 104.243-4.242 3 3 0 00-4.243 4.242z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
  <h3
    class="mt-2 text-base font-bold text-gray-100"
  >
    Do a search!
  </h3>
  <p
    class="mt-1 text-sm text-gray-400"
  >
    Search for a movie above, or try any of the suggestions below:
  </p>
  <div
    class="mt-6 w-full text-center lg:max-w-[500px]"
  >
    <span
      class="inline-flex items-center py-0.5 px-2.5 mx-1 text-xs font-medium text-gray-800 bg-gray-100 hover:bg-gray-300 rounded-full"
    >
      <a
        href="/?search=Minions"
      >
        Minions
      </a>
    </span>
    <span
      class="inline-flex items-center py-0.5 px-2.5 mx-1 text-xs font-medium text-gray-800 bg-gray-100 hover:bg-gray-300 rounded-full"
    >
      <a
        href="/?search=Captain%20Marvel"
      >
        Captain Marvel
      </a>
    </span>
    <span
      class="inline-flex items-center py-0.5 px-2.5 mx-1 text-xs font-medium text-gray-800 bg-gray-100 hover:bg-gray-300 rounded-full"
    >
      <a
        href="/?search=Rambo"
      >
        Rambo
      </a>
    </span>
    <span
      class="inline-flex items-center py-0.5 px-2.5 mx-1 text-xs font-medium text-gray-800 bg-gray-100 hover:bg-gray-300 rounded-full"
    >
      <a
        href="/?search=Mission%20Impossible"
      >
        Mission Impossible
      </a>
    </span>
    <span
      class="inline-flex items-center py-0.5 px-2.5 mx-1 text-xs font-medium text-gray-800 bg-gray-100 hover:bg-gray-300 rounded-full"
    >
      <a
        href="/?search=Spiderman"
      >
        Spiderman
      </a>
    </span>
  </div>
</div>
`;
